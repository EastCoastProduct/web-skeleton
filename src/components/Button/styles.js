import { StyleSheet } from 'aphrodite/no-important';

export default StyleSheet.create({
  empty: {
    background: 0,
    border: 0,
    color: 'inherit',
    outline: 'none',
    padding: 0,
  },
});
