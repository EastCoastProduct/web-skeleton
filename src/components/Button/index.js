import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './styles';

const Button = ({ children, empty, ...rest }) =>
  <button
    className={css(empty && styles.empty)}
    {...rest}
  >{children}</button>;

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]).isRequired,
  empty: PropTypes.bool,
};

export default Button;
