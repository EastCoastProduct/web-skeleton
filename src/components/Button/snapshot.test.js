import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheet, StyleSheetTestUtils } from 'aphrodite/no-important';
import Button from './';

describe('Button component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data', () => {
    const tree = renderer.create(<Button>Login</Button>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders empty button', () => {
    const tree = renderer.create(<Button empty>Login</Button>);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
