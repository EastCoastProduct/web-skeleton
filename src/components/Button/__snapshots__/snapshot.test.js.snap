exports[`Button component snapshot renders basic required data 1`] = `
<button
  className="">
  Login
</button>
`;

exports[`Button component snapshot renders empty button 1`] = `
<button
  className="empty_1b6lvt9">
  Login
</button>
`;
